mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -6316,6 +6316,20 @@ OpFoldResult SplatOp::fold(FoldAdaptor adaptor) {
   return SplatElementsAttr::get(getType(), {constOperand});
 }
 
+//===----------------------------------------------------------------------===//
+// StepOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult StepOp::fold(FoldAdaptor adaptor) {
+  auto resultType = cast<VectorType>(getType());
+  if (resultType.isScalable())
+    return nullptr;
+  SmallVector<APInt> indices;
+  for (unsigned i = 0; i < resultType.getNumElements(); i++)
+    indices.push_back(APInt(/*width=*/64, i));
+  return DenseElementsAttr::get(resultType, indices);
```

Comment:
Why do we want to canonicalize the fixed-length variant in the Vector dialect to something else that is not a `vector.step`? I'd suggest that we use `vector.step` as canonical representation for both fixed and scalable versions for consistency. Is there any reason to not doing so?

Reply:
Purely for convenience to be honest, this seemed like the simplest way to add the op. I know the downside is information is lost when folding to a constant (i.e. it's a constant step of one), as well as consistency as you point out, so am happy to iterate on this if folks would like to see this done differently.

Reply:
Thanks for clarifying! I think consistency is really important to make sure fixed and scalable worlds are aligned when possible and we don't end up with a different set of patterns and problems when we shouldn't. `vector.step` looks like a good abstraction for both fixed and scalable vectors so I'm +1 to removing this canonicalization pattern and let `vector.step` be the canonical representation also for fixed vectors.